### PR TITLE
Cycle 4: Moritz to work on `astropy.io.ascii` and X-ray spectral modelling

### PR DESCRIPTION
**Pull request by hamogu**
### Project Description

Address (by fixing or closing) open issues in `astropy.io.ascii`

### Project / Work

At the time of writing there are [66 open issues in astropy tagged `io.ascii`](https://github.com/astropy/astropy/issues?q=is%3Aopen+is%3Aissue+label%3Aio.ascii), 57 of which are older than 18 months.

While astropy clearly thrives with those bugs and issues, each of them represents at least one user
who was stopped in their work and found this annoying enough to open an issue. We know anecdotally, that
many users who do not think of themselves as "developers" don't open issues on github (e.g. they might not 
even have an account) so each bug probably represents several or more users who have run
into a problem. For a smooth user experience, we should close out bugs and fill in feature requests
where we can.

Not all of these issues are actionable (e.g. some 
require upstream fixes or 
[would require fairly invasive changes that would break backwards compatibility in a way that does not seem warrented](https://github.com/astropy/astropy/issues/5440#issuecomment-880703756));
however most can be addressed given some developer time.

Moritz proposes to pay a sub-grant to MIT to pay for some of his time to
work on those issues. Based on previous PRs, I estimate that I can address
on average one issue in one to two work days, so paying for 10% of my work time for one year should allow me to address the majority of the currently open issues.

### CODE REVIEW

**Comment by kelle** [comment]
- Since this is a subaward and is to do work which is likely to extend for more than one year (`io.ascii` maintenance), please consider submitting a budget for 2-3 years instead of just one.  
- This request could also mention your other efforts in the Project.

**Comment by hamogu** [comment]
I do not anticipate working on this for more than one year at the level of effort I'm asking for. There are about 50 open issues on `io.ascii`. I'm not qualified to address all of them (e.g. in the fast reader in C code) and some issues are not actionable, e.g. they are asking for things so ill-defined that it cannot be implemented. One year is probably enough to address the issues that can be addressed by me.

Re: "could also mention my other efforts in the project": I was planning to continue my other work as unfunded volunteer. This proposal is to pay for time for astropy that I would not have otherwise, not to get me funded through astropy for things I'm doing anyway. However, I realize that it might not be clear to others "what I would be doing anyway" so I'll try to broaden the language to include any aspects where ambiguity might arise.

**Comment by hamogu** [comment]
The release of the president's budget request two days ago indicates that NASA intends to wind down Chandra operations as soon as possible. While that's only a request and not an approved budget and might still change, it is possible (or even likely) that I won't be able to use my "science time" as paid for by Chandra to volunteer much longer. I'm thus following @kelle suggestion to broaden the scope so that, if granted, this could also be used to pay for my other roles on astropy and I'm suggesting a a possible path to a 2-3 year time line.

File: finance/proposal-calls/cycle4/moritz_ascii.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Addressing issues in `astropy.io.ascii`
+
+### Project Team
+
+- Moritz Guenther (@hamogu)
+
+### Project Description
+
+Address (by fixing or closing) open issues in `astropy.io.ascii`
```
Comment (hamogu) [inline finance/proposal-calls/cycle4/moritz_ascii.md:11]:
```suggestion
Address (by fixing or closing) open issues in `astropy.io.ascii`, astropy affiliate editor, organization of dev-telecons, development and review of other astropy and specutils code.
```

File: finance/proposal-calls/cycle4/moritz_ascii.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+### Project / Work
+
```
Comment (hamogu) [inline finance/proposal-calls/cycle4/moritz_ascii.md:15]:
```suggestion
Part 1: `astropy.io.ascii`
```

File: finance/proposal-calls/cycle4/moritz_ascii.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+many users who do not think of themselves as "developers" don't open issues on github (e.g. they might not 
+even have an account) so each bug probably represents several or more users who have run
+into a problem. For a smooth user experience, we should close out bugs and fill in feature requests
+where we can.
+
+Not all of these issues are actionable (e.g. some 
+require upstream fixes or 
+[would require fairly invasive changes that would break backwards compatibility in a way that does not seem warrented](https://github.com/astropy/astropy/issues/5440#issuecomment-880703756));
+however most can be addressed given some developer time.
+
```
Comment (hamogu) [inline finance/proposal-calls/cycle4/moritz_ascii.md:29]:
```suggestion

Part 2:
Organizational work for Astropy:
- Astropy affiliate package editor with pyOpenSci
- organization of dev-telecons
- strategic planning and proposal writing

Part 3:

Bring X-ray spectroscopy to `specutils` and `astropy.modelling`. Moritz is one of the core developers of [sherpa](https://github.com/sherpa/sherpa/). Sherpa is a complete package for X-ray spectral modelling with an infrastructure very similar to `astropy.modelling`, providing (a) some models and fitters that astropy misses and (b) ways to handle [spectra and responses in the OGIP format](https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/spectra/ogip_92_007/node5.html). Indeed, we have made a [bridge between sherpa and astropy a few years ago](https://saba.readthedocs.io/en/latest/), but maintaining compatibility between three separate packages (astropy, saba, sherpa) through versions turned out to be cumbersome. Also, Sherpa has a lot of infrastructure to read OGIP compliant X-ray data (PHA, ARF, RMF) that would fit well into a `specutils.OGIP` package to make those formats available to other parts of the astropy ecosystem. With Sherpa's continued developent in question given NASA's budget proposal to wind down Chandra support, I suggest to move or re-implement (depending on package license) those routines. Several (non affiliated) Python packages have done bits and parts of this, but it seems the time has come to bring the OGIP formats to a well-used astropy ecosystem package like `spectuils`.
```

File: finance/proposal-calls/cycle4/moritz_ascii.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Budget breakdown (nominal):
+
+- $25,000 for 10% of my time for 1 year
+
+(Since this will be a sub-award to MIT, my salary is fixed by MIT and the overhead rate is fixed by agreement between MIT and the US government. We can change the length of the contract, but not the hourly rate.)
+
+The minimum useful time is determined by how much effort it is to set up that sub-award. I can work on one issue in 8 hours of work, but neither MIT nor NumFOCUS would want to write and process a subaward contract for that little money. Realistically, it's probably not worth the effort if less than $5-10k  are allocated to this project.
+
+### Period of Performance
+1 year
```
Comment (hamogu) [inline finance/proposal-calls/cycle4/moritz_ascii.md:47]:
```suggestion
3 years
```

File: finance/proposal-calls/cycle4/moritz_ascii.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+work on those issues. Based on previous PRs, I estimate that I can address
+on average one issue in one to two work days, so paying for 10% of my work time for one year should allow me to address the majority of the currently open issues.
+
+
+
+### Approximate Budget
+
+Budget breakdown (nominal):
+
+- $25,000 for 10% of my time for 1 year
```
Comment (hamogu) [inline finance/proposal-calls/cycle4/moritz_ascii.md:40]:
```suggestion
- $25,000 for 10% of my time per year
```

**Comment by hamogu** [comment]
@mhvk : You already gave a "thumb up". Note that the comment by @kelle  and the release of the president's budget request two days ago have let me to significantly increase the scope for this proposal to 3 years.

**Comment by mhvk** [comment]
Definitely another :+1: to the idea to bring the best from X-ray spectral modeling to `specutils`! It will be nice to have it in a package aimed at all wavelengths: modeling multi-band photometry is not all that different from modelling X-ray CCD spectra. I also more generally like seeing proposals to add missing pieces of functionality! (Though your "grunt" work is always very much appreciated too, of course!)

**Comment by jdswinbank** [comment]
While obviously we'd love to fund this for the full amount listed, if we need to squeeze the budget is there a minimum number of hours per year that would make sense?

**Comment by hamogu** [comment]
The minimum is defined here: https://github.com/astropy/astropy-project/pull/385/files#diff-15111b979d86e382215dfe1ee880579cbc96ce543e042c956d8c166bb7a68237R53

Copying that line below for convenience:
The minimum useful time is determined by how much effort it is to set up that sub-award. I can work on one issue in 8 hours of work, but neither MIT nor NumFOCUS would want to write and process a subaward contract for that little money [8h /year]. Realistically, it's probably not worth the effort if less than $5-10k  are allocated to this project.

**Comment by jdswinbank** [comment]
Thanks Moritz; dunno how I missed that first time!

**Comment by eteq** [comment]
Please react to this comment to vote on this proposal (👍, 👎, or no reaction for +0)

**Comment by kelle** [comment]
I'm writing on behalf of Astropy's Finance Committee regarding the outcome of this funding request. We are pleased to be able to let you know that, following consultation with the community, we are able to approve this request. We can currently fund the Year 1 amount of US$ 5000 to carry out the project. Funding beyond this amount will be contingent upon the availability of funds. (We will be using the full budgets of all of the approved requests to craft future grant and funding proposals.) We assume you will be using this full year 1 budget, if that is not the case please contact us immediately.

Ana Gabela and I will be your contacts on the Finance Committee to facilitate this award. Please get in touch with us if you have any questions or concerns. Please do not reach out to NumFOCUS directly. Also, after merging this PR, I will open an issue where we can discuss progress on this project.

In addition, new to this funding cycle, is the assignment of a COTR (Contracting Officer's Technical Representative) to each funded project. This concept is borrowed from government funding agencies, although it is to be stressed that Astropy's goal is to make the COTR role as low-overhead as possible. The COTR’s primary responsibility is to make sure the work is happening at the expected pace and, if necessary, to be a liaison between the funded project and the Finance Committee or CoCo. The COTR for your project will be assigned shortly and we’ll also be sending out more details about how we see this working.

We have already been in touch with both NumFOCUS and MIT on getting this subaward setup. 

Congratulations --- we are really looking forward to seeing you put these funds to good use!

Kelle, on behalf of the Astropy Finance Committee

**Pull request merged**